Sprint 19

ZakharVP opened the pull request:
Картинку "Ошибка сервера" расположил на экране настроек
Картинку "Нет интернета расположил на экране информация по перевозчику.

Commit:
Сделал задание по 19 спринту

Commit:
Убран файл

ValeriUst commented on travel_go/Views/MainScreen.swift:
```diff
@@ -0,0 +1,27 @@
+    var body: some View {
+        TabView(selection: $selectedTab) {
+            ChoiceRoute()
+                .tabItem { Label("", image: "MainIcon") }
```
Можешь еще изучить тоже удобный вариант генерации картинок и цветов - [Asset symbol generation in Xcode 15](https://sarunw.com/posts/swift-symbols-for-asset-catalog/) — это делает работу с ними безопаснее и удобнее.

ValeriUst commented on travel_go/Views/MainScreen.swift:
```diff
@@ -0,0 +1,27 @@
+import SwiftUI
+
+struct MainScreen: View {
+    @State var selectedTab: Int = 0
```
@State всегда должен быть приватным

ValeriUst commented on travel_go/Views/Other/CarrierDetail.swift:
```diff
@@ -0,0 +1,31 @@
+                .frame(width: 223, height: 223)
+            
+            // Надпись под картинкой
+            Text("Нет интернета")
```
Строки лучше выносить в приватные enum

ValeriUst commented on travel_go/Views/Other/CarrierDetail.swift:
```diff
@@ -0,0 +1,31 @@
+            Image("no_internet")
+                .resizable()
+                .scaledToFit()
+                .frame(width: 223, height: 223)
```
223 -  [магические числа](https://refactoring.guru/ru/replace-magic-number-with-symbolic-constant) Лучше выносить такие значения в свойства или в приватный enum с понятным названием.

ValeriUst commented on travel_go/Views/Other/ConsoleScreenView.swift:
```diff
@@ -1,18 +1,19 @@
-struct ContentView: View {
+struct ConsoleScreenView: View {
+   
     @State private var copyrightInfo: String = "Загрузка..."
```
Это хорошо

ValeriUst commented on travel_go/Views/SomeChoice/ChoiceTime.swift:
```diff
@@ -0,0 +1,27 @@
+    
+    var body: some View {
+        NavigationView {
+            Form {
```
В таком случае рекомендую добавлять тудушку

ValeriUst commented on travel_go/Views/Tabs/ChoiceRoute.swift:
```diff
@@ -0,0 +1,158 @@
+    }
+
+    private func findRoutes() {
+        print(
```
Перед сдачей работы рекомендуется удалить print-ы

ValeriUst commented:
⚠️ Критические замечания (это нужно исправить обязательно):

1. Приложение в тёмной теме не соответствует макетам — требуется привести экраны в соответствие с Figma-дизайном (Пункут 10).
<img width="1179" height="2556" alt="IMG_9251" src="https://github.com/user-attachments/assets/f2e422a8-67d6-4d99-9f02-8a5ac537fc82" />
<img width="1179" height="2556" alt="IMG_9252" src="https://github.com/user-attachments/assets/f6813772-bfd5-441d-bae8-c93644a2a74f" />
<img width="1179" height="2556" alt="IMG_9253" src="https://github.com/user-attachments/assets/d077f4c4-5ef1-441e-a483-17e10b689c41" />
<img width="1179" height="2556" alt="IMG_9254" src="https://github.com/user-attachments/assets/de4ca3d5-3031-485d-838b-3ed0c9f1b171" />

Commit:
fix: Исправление темной темы

Исправил использование фона и цвет текста на адаптивные цвета
Добавил синему цвету такое же отражение в темной теме

Commit:
fix: исправление приватного модификатора

Исправил на MainScreen модификатор @State

Commit:
fix: Изменил использование строк на enum

Сделал отдельный файл для строк и в поле Text использую значение констант

Commit:
fix: Магические числа

Убрал значения картинки в константы

Commit:
fix: Print для отладки

Убрал из кода со страниц отладачные принты
